pkg/controller/rest/didexchange/models.go

```diff
@@ -201,14 +201,27 @@ type RemoveConnectionRequest struct { // nolint: unused,deadcode
 type RemoveConnectionResponse struct { // nolint: unused,deadcode
 }
 
-// saveConnectionResp model
+// createConnectionResp model
 //
 // This is used as the response model for save connection api.
 //
-// swagger:response saveConnectionResp
-type saveConnectionResp struct { // nolint: unused,deadcode
+// swagger:response createConnectionResp
+type createConnectionResp struct { // nolint: unused,deadcode
 	// The ID of the connection to get
 	//
 	// in: body
 	ID string `json:"id"`
 }
+
+// createConnectionRequest model
+//
+// Request to create a new connection between two DIDs.
+//
+// swagger:parameters createConnection
+type createConnectionRequest struct { // nolint: unused,deadcode
+	// Params for creating a connection.
+	//
+	// in: body
+	// required: true
+	Request didexchange.CreateConnectionRequest
```

Comment:
Should be:
```
/// in: body
Body struct {
	// required: true
	Request didexchange.CreateConnectionRequest
}
```

Reply:
@soluchok can you give an example of the swagger output you're looking for?

This is the current output:

```yaml
  /connections/create:
    post:
      operationId: createConnection
      parameters:
      - description: Params for creating a connection.
        in: body
        name: Request
        required: true
        schema:
          $ref: '#/definitions/CreateConnectionRequest'
      responses:
        "200":
          $ref: '#/responses/createConnectionResp'
        default:
          $ref: '#/responses/genericError'
      summary: Saves the connection record.
      tags:
      - did-exchange
```

```yaml
  CreateConnectionRequest:
    description: CreateConnectionRequest model
    properties:
      implicit:
        type: boolean
        x-go-name: Implicit
      invitationDID:
        type: string
        x-go-name: InvitationDID
      invitationID:
        type: string
        x-go-name: InvitationID
      myDID:
        type: string
        x-go-name: MyDID
      parentThreadID:
        type: string
        x-go-name: ParentThreadID
      theirDID:
        $ref: '#/definitions/DIDDocument'
      theirLabel:
        type: string
        x-go-name: TheirLabel
      threadID:
        type: string
        x-go-name: ThreadID
    type: object
    x-go-package: github.com/hyperledger/aries-framework-go/pkg/controller/command/didexchange

```

Reply:
Note this other example from the same file:

https://github.com/hyperledger/aries-framework-go/blob/27f435692fd574ab373e76f340d383b80c9a2a9e/pkg/controller/rest/didexchange/models.go#L31-L39

Reply:
Sorry, my fault. I thought we need a `Request` field as well.
eg. 
```
{
"myDID": "string"
...
}
vs
{
    "request":{
        "myDID": "string"
        ...
    }
}
```
But `createConnectionResp` is definitely wrong. Should be:
```
type createConnectionResp struct { // nolint: unused,deadcode
	// The ID of the connection to get
	//
	// in: body
	Body struct {
		ID string `json:"id"`
	}
}
```

Reply:
> Note this other example from the same file:
> 
> https://github.com/hyperledger/aries-framework-go/blob/27f435692fd574ab373e76f340d383b80c9a2a9e/pkg/controller/rest/didexchange/models.go#L31-L40

it is wrong. it will show only the `didexchangeSvc.Invitation` structure. All tags will be ignored `alias , invitation, invitation_url `

Reply:
@soluchok 



> > Note this other example from the same file:
> > https://github.com/hyperledger/aries-framework-go/blob/27f435692fd574ab373e76f340d383b80c9a2a9e/pkg/controller/rest/didexchange/models.go#L31-L40
> 
> it is wrong. it will show only the `didexchangeSvc.Invitation` structure. All tags will be ignored `alias , invitation, invitation_url `

Verified:

```yaml
  /connections/create-invitation:
    post:
      operationId: createInvitation
      parameters:
      - description: The Alias to be used in invitation to be created
        in: query
        name: alias
        type: string
        x-go-name: Alias
      - description: Optional public DID to be used in invitation
        in: query
        name: public
        type: string
        x-go-name: Public
      responses:
        "200":
          $ref: '#/responses/createInvitationResponse'
        default:
          $ref: '#/responses/genericError'
      summary: Creates a new connection invitation....
      tags:
      - did-exchange
```

```yaml
  createInvitationResponse:
    description: |-
      createInvitationResponse model

      This is used for returning a create invitation response with a single connection invitation as body
    schema:
      $ref: '#/definitions/Invitation'
```

Reply:
@soluchok 

> But `createConnectionResp` is definitely wrong. Should be:
> 
> ```
> type createConnectionResp struct { // nolint: unused,deadcode
> 	// in: body
> 	Body struct {
> 	       // The ID of the connection to get
> 	       //
> 		ID string `json:"id"`
> 	}
> }
> ```

Fixed. New swagger output:

```yaml
  createConnectionResp:
    description: |-
      createConnectionResp model

      This is used as the response model for save connection api.
    schema:
      properties:
        id:
          description: The ID of the connection to get
          type: string
          x-go-name: ID
      type: object
```